### DISCUSSION TITLE
logfile output for cell-mechanics

### DISCUSSION THREAD

**Discussion by cerice1995** [UvaCsl/HemoCell#40] [General discussion]
In the default RBC.xml file, k_link is 15 and k_bend is 80, but different numbers are printed in the logfile. How are the logfile numbers generated? What are the units (when not dimensionless)? 

Also, I noticed that the stretchCell logfile and the oneCellShear logfile values are not the same. Is something going wrong, or is this the expected output? 

-------

From stretchCell log:
  (Cell-mechanics model) High Order model parameters for RBC cellfield
           k_link:   0.00128017
           k_area:   6.40083e-06
           k_bend: : 0.000102413
           k_volume: 2.56033e-05
           eta_m:    0
           mean_edge:1.00019
           N faces:  1280

From oneCellShear:
  (Cell-mechanics model) High Order model parameters for RBC cellfield
           k_link:   0.000320041
           k_area:   1.60021e-06
           k_bend: : 2.56033e-05
           k_volume: 6.40083e-06
           eta_m:    0
           mean_edge:1.00019
           N faces:  1280

**Comment by cerice1995** [comment]
I found these calculations in cellMechanics.h. Variable dx is 5e-07 in both logfiles, and while I don't see kBT or kBT_lbm recorded in the logfile, my initial thought is that the examples are at the same kBT (T = 300 or kBT =4.100531391e-21). 

----

T calculate_kLink(Config & cfg, plb::MeshMetrics<T> & meshmetric){
    T kLink = cfg["MaterialModel"]["kLink"].read<T>();
    T persistenceLengthFine = 7.5e-9; // In meters -> this is a biological value
    T plc = persistenceLengthFine/param::dx; //* sqrt((meshmetric.getNumVertices()-2.0) / (23867-2.0)); //Kaniadakis magic
    return  kLink * param::kBT_lbm/plc;
  };
  
  T calculate_kBend(Config & cfg, plb::MeshMetrics<T> & meshmetric ){
    T eqLength = 5e-7/param::dx;
    return cfg["MaterialModel"]["kBend"].read<T>() * param::kBT_lbm / eqLength;
  };

**Comment by gzavo** [comment]
This is the expected behavior. The mechanical model parameters are printed in lattice units. While they are the same in SI, if dx or dt is different between the two simulations (and the latter is likely), then the lattice unit values will be different.
